src/components/FileList.jsx

```diff
@@ -83,19 +83,22 @@ class FileList extends PureComponent {
 
   rowRenderer = ({ index, key, style }) => {
     const {
-      displayedFolder, selected, selectionModeActive, onFileEdit, onFolderOpen, onFileOpen, onFileToggle, showActionMenu
+      displayedFolder, selected, selectionModeActive, onFolderOpen, onFileOpen, onFileToggle, showActionMenu
     } = this.props
     const file = this.props.files[index]
     if (!file) {
       return <FilePlaceholder key={key} style={style} />
     }
+    const { isRenaming, renamingFile } = this.props
+    const isFileRenaming = isRenaming && renamingFile && renamingFile.id === file.id
+    const isSelected = selected.find(f => f && f.id === file.id) !== undefined
     return (
       <File
```

Comment:
I know it is not this PR's purpose, but I would like to highlight that this component **smells a lot**: too many attributes makes the component hard to test and it bears too many responsibilities.